django/test/utils.py

```diff
@@ -94,13 +94,23 @@ def instrumented_test_render(self, context):
     return self.nodelist.render(context)
 
 
-def setup_test_environment():
+class _SavedSettings(object):
+    pass
+
+
+def setup_test_environment(debug=None):
     """Perform any global pre-test setup. This involves:
 
         - Installing the instrumented test renderer
         - Set the email backend to the locmem email backend.
         - Setting the active locale to match the LANGUAGE_CODE setting.
     """
+    if debug is None:
```

Comment:
It would be nice to test this condition. Can we safely do that in Django's test suite? Could a test safely call `teardown_test_environment()` first to avoid the nested calls issue?